### DISCUSSION TITLE
How can we change the default icons of the providers?

### DISCUSSION THREAD

**Discussion by dorutuu** [add2cal/add-to-calendar-button#586] [Q&A]

**Comment by dorutuu** [comment]
I really want to change the apple logo into the Calendar icon from the iPhone or Mac

**Comment by dorutuu** [comment]
I'm using the react component if helps 😃

**Comment by jekuer** [comment]
Unfortunately, there is no easy way to do this at the moment.
We are considering moving the icons into the css part in the future - this would make it easier adjustable.
For now, please have a look at #421 - there, we highlight a potential workaround to add your own icon or similar.

**Reply by dorutuu** [reply]
Doesn't seem to be working for me, I copied the exact code, but we I set the route for the customCss the button simply disappears

**Reply by jekuer** [reply]
When using the customCss option, did you specify the buttonStyle to "custom" and is the customCss file a copy of the one in the repo, but altered?
What would not work: only putting selected css into this customCss option, as it replaces the css - it is not adding css!

**Reply by dorutuu** [reply]
I added the buttonStyle to "custom" and it's the same result. 

I just want the same functionality only with a different icon. 

I can use the react one or the other one, doesn't matter. 
Thanks a lot in advance 🙏🏻

**Reply by dorutuu** [reply]
Sorry if none of this made sense, after I click on the button my code renders a modal like this:  
![image](https://github.com/add2cal/add-to-calendar-button/assets/130478504/5f951f13-f692-4867-b574-1b1a30d3b32c)

How can I change the default Apple logo from there? can I overwrite it somehow, I've tried everything so far and none of it work

**Reply by jekuer** [reply]
Can you please share your source code or a link to a page (or demo) where you have implemented this - as this is far away from the default?

**Reply by dorutuu** [reply]
https://codesandbox.io/p/sandbox/add-to-calendar-q3lwhm?file=%2Fsrc%2FApp.tsx

Let me know if you can see it

I was thinking if you gave a id or a class to the svg I believe I can ovewrite it with mask-image or something, but that's just a thought. 

Thanks a lot

**Reply by jekuer** [reply]
Alright. As this is not officially supported at the moment, I cannot really support here in detail.
I would recommend you build the modal on your own and use individual buttons per type (as you did on the main level already).
We are planning to optimize here in the near future - however, near won't be in the next 2 months.

You can get to a hacky solution by using a custom style, where you infuse the icon ::before the option-text container, while hidding the actual icon via the respective official option. However, as said, no official solution; will take some time and sweat to get this running. Would not recommend it.
